src/layers/basic.jl

```diff
@@ -503,3 +503,54 @@ function Base.show(io::IO, e::Embedding)
 end
 
 outputsize(e::Embedding) = (e.out_dims,)
+
+"""
+    Periodic(dims, periods)
+
+Create a layer periodic in `dims` with respective periods `periods`. All other input
+dimensions are passed through unchanged, but `dims` are replaced with their sine and cosine
+(scaled appropriately to have the specified periods).
+
+## Arguments
+
+  - `dims`: Periodic dimensions
+  - `periods`: Periods of the dimensions `dims`, in the same order
+
+## Inputs
+
+  - `x` must be an AbstractArray with `issubset(dims, axes(x, 1))`
+
+## Returns
+
+  - AbstractArray with dimensions `(size(x, 1) + length(dims), ...)` where `...` are the
+    other dimensions of `x`; the unchanged dimensions are first, then all the sines, then
+    all the cosines.
+  - Empty `NamedTuple()`
+"""
```

Comment:
Let's add a doctest here. From the docstring it is not very obvious what is happening.

Reply:
I've updated the docstring and added a doctest. Please let me know if it's clear now. I think that was also the last edit needed for all the tests to pass, so I believe it's ready to review. Thanks for all your help with this.